LoginProvider.swift

```diff
@@ -0,0 +1,66 @@
+//
+//  LoginProvider.swift
+//  Authentication
+//
+//  Created by Gustavo Cairo on 1/13/17.
+//  Copyright © 2017 Wolox. All rights reserved.
+//
+
+import Foundation
+import ReactiveCocoa
+import ReactiveSwift
+import enum Result.NoError
+
+/**
+    Protocol that all users returned by
+    login providers should implement.
+ */
+public protocol LoginProviderUser {}
+
+/**
+    Protocol that all login provider configurations
+    should implement.
+ */
+public protocol LoginProviderConfiguration {}
+
+/**
+    This enum lists all the possible user types
+    login providers can return.
+ 
+    If a service you plan to use is not listed,
+    you may use `custom` as your user type, and
+    name it.
+ */
+public enum LoginProviderUserType {
+    case facebook(user: FacebookLoginProviderUser)
+    case custom(name: String, user: LoginProviderUser)
+}
+
+/**
+    Protocol that all LoginProviders must implement.
+*/
+public protocol LoginProvider {
+    
+    /**
+        Signal that returns a user created with the LoginProvider.
```

Comment:
`Signal that returns the user created when the login was triggered in the LoginProvider.` ?